crypto/bio/bf_lbuf.c

```diff
@@ -235,9 +235,7 @@ static long linebuffer_ctrl(BIO *b, int cmd, long num, void *ptr)
         obs = (int)num;
         p = ctx->obuf;
         if ((obs > DEFAULT_LINEBUFFER_SIZE) && (obs != ctx->obuf_size)) {
-            if (num <= 0)
-                return 0;
-            p = OPENSSL_malloc((size_t)num);
+            p = OPENSSL_malloc((size_t)obs);
```

Comment:
I think this might have been one of the recent fixes for passing a negative value to malloc().  I don't see a path where `num` or `obs` is checked for being >0.
Also, changing `num` to `obs` risks losing information by truncation (`long` to `int`).

Reply:
?? if (num <= 0) was doing that check

Reply:
Just look at the condition the block is in: `(obs > DEFAULT_LINEBUFFER_SIZE)`

And the truncation is irrelevant (or rather it should be reported as an error but that is a different issue) as we already handle everything in this BIO within the int type size.

Reply:
I can add the truncation check `if (num > INT_MAX) return 0;` if you want.

Reply:
Safest to have it.

Reply:
fixup pushed